test/common_device_type.py

```diff
@@ -229,6 +278,60 @@ def only_fn(slf, device, *args, **kwargs):
         return only_fn
 
 
+# Lists of Torch types
+DOCUMENTED_TENSOR_TYPES = [torch.float64, torch.float32, torch.float16,
+                           torch.int64, torch.int32, torch.int16, torch.int8,
+                           torch.uint8, torch.bool]
+
+# SCALAR_TYPES = DOCUMENTED_TENSOR_TYPES - half and bool
+# Note: derived from AT_FORALL_SCALAR_TYPES
+SCALAR_TYPES = [torch.float64, torch.float32,
+                torch.int64, torch.int32, torch.int16, torch.int8,
+                torch.uint8]
+
+QUANTIZED_TYPES = [torch.quint8, torch.qint8, torch.qint32]
+
+ALL_TYPES = [torch.bfloat16] + DOCUMENTED_TENSOR_TYPES + QUANTIZED_TYPES
```

Comment:
To start, I'd just have one or two lists corresponding to:
https://github.com/pytorch/pytorch/blob/98ccae09afc8e1d67ff62b3e6fd0ecec21328993/aten/src/ATen/Dispatch.h#L174-L190
and https://github.com/pytorch/pytorch/blob/98ccae09afc8e1d67ff62b3e6fd0ecec21328993/aten/src/ATen/Dispatch.h#L240-L254

If there is some (understandable) queasiness around calling that "ALL" we can call it "STANDARD" or "COMMON" or something and deprecate the dispatch macro with the "ALL" name.

I don't think an actual "ALL" approach is useful here -- you can't even construct e.g. quantized dtypes with the same constructor as non-quantized dtypes, so there's no point in writing a generic test.

Reply:
I agree with you. Let's just keep a SCALAR_TYPES list (like the one that exists) that corresponds to AT_DISPATCH_ALL_TYPES (which I think the current one does)? 

I don't think we need the "AND" variation because you can just + the dtypes you want in Python. 

`@dtypes(SCALAR_TYPES + [torch.float16])`

Reply:
> SCALAR_TYPES

I'm not a huge fan of that name because it's a C++ concept but we are in the python land.  I'd prefer something like STANDARD_DTYPES, COMMON_DTYPES or whatever.

> I don't think we need the "AND" variation because you can just + the dtypes you want in Python.

Ya I thought about that, it was just to make it more clear that the names correspond to the dispatch macros.  But if you want to just comment that fact instead, that sounds good to me.

Reply:
Sure. Let's either remove it or name it COMMON_DTYPES.